src/main/java/com/google/sps/servlets/DatastoreMetadataStore.java

```diff
@@ -0,0 +1,121 @@
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DatastoreMetadataStore implements MetadataStore {
+
+  public DatastoreMetadataStore() { }
+
+  /**
+   * Stores a valid Metadata type into Datastore
+   */
+  public void storeData(Metadata data) {
+    // Create service call to access Datastore
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    // Create a new Entity
+    Entity datasetEntity = new Entity("MetaData");
+    datasetEntity.setProperty("user-email", data.getUser());
+    datasetEntity.setProperty("dataset-name", data.getDataset());
+    datasetEntity.setProperty("model", data.getModel());
+    datasetEntity.setProperty("visualizer-type", data.getVisualization());
+    datasetEntity.setProperty("image-count", data.getNumberOfImages());
+    datasetEntity.setProperty("timestamp", data.getTimestamp());
+
+    // Place entity withing Datastore
+    datastore.put(datasetEntity);
+  }
+  
+  /**
+   * Retrieves the metadata information stored withing Datastore
+   * using a given dataset name.
+   */
+  public Metadata retrieveMetadata(String datasetName) {
+    // Check to see if any entity has the same matching dataset name
+    for (Entity entity : getAllDatastoreEntities().asIterable()) {
+      String dataset = (String) entity.getProperty("dataset-name");
+
+      if (dataset.equals(datasetName)) {
+        // Get the remaining properties
+        String email = (String) entity.getProperty("user-email");
+        String model = (String) entity.getProperty("model");
+        String visualizerType = (String) entity.getProperty("visualizer-type");
+        long imageCount = (Long) entity.getProperty("image-count");
+        long time = (Long) entity.getProperty("timestamp");
+        return new Metadata(email, dataset, model, visualizerType, imageCount, time);
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Retrieves all the names of datasets belonging to a 
+   * particular user.
+   */
+  public List<String> getUsersDatasets(String user) {
+    List<String> userDatasets = new ArrayList<String>();
```

Comment:
Do you need to get all the datastore entities and filter manually? I think the Query has a filter based on user-email or dataset-name.